### ISSUE TITLE
nan values while training

### ISSUE THREAD

**Issue by m7schoen** [HRLTY/TP-GAN#3]
I managed to start training on the sample data. However after a couple of batches all the lossess instantly turn to nan and the saved output images become black. I'm using python3 and tensorflow 1.4.1. Any ideas?

**Comment by HRLTY** [comment]
Hi. Might comes from the imbalanced data for G and D, since GAN is not stable especially initially.
Rui

**Comment by m7schoen** [comment]
Hi, that was also my suspicion. So I started to remove components like the discriminator and also the lcnn. Still the same issue. Now I have basically only the "g_dec_optim" operation left.

````
... Epo[ 0][   2/ 771][t24.60] d_l:0.0000 fstol:11031 cg32:0 cg64:0 cg128:0 el:714 er:1141 no:848 mo:974
g_l:13325 gL1:7938(sym:223) gL2:2202(sym:5060) gL3:580(sym:1380) gadv:0.0000 dv_l:0.00 iden:10.7992, tv:14698 

.......... Epo[ 0][  12/ 771][t33.60] d_l:0.0000 fstol:12138 cg32:0 cg64:0 cg128:0 el:761 er:1181 no:936 mo:1168 g_l:13926 gL1:8829(sym:229) gL2:2095(sym:4770) gL3:545(sym:1256) gadv:0.0000 dv_l:0.00 iden:11.9658, tv:14429 

.......... Epo[ 0][  22/ 771][t42.60] d_l:0.0000 fstol:10831 cg32:0 cg64:0 cg128:0 el:706 er:1069 no:805 mo:1030 g_l:12484 gL1:7741(sym:242) gL2:1854(sym:4481) gL3:521(sym:1217) gadv:0.0000 dv_l:0.00 iden:14.4117, tv:14046 

.......... Epo[ 0][  32/ 771][t51.64] d_l:0.0000 fstol:11494 cg32:0 cg64:0 cg128:0 el:756 er:1116 no:875 mo:1084 g_l:13542 gL1:8844(sym:248) gL2:1844(sym:4449) gL3:495(sym:1186) gadv:0.0000 dv_l:0.00 iden:16.8582, tv:13815 

.......... Epo[ 0][  42/ 771][t60.70] d_l:0.0000 fstol:12165 cg32:0 cg64:0 cg128:0 el:844 er:1164 no:894 mo:1154 g_l:13800 gL1:9547(sym:210) gL2:1676(sym:4090) gL3:464(sym:1064) gadv:0.0000 dv_l:0.00 iden:11.2798, tv:13617 

.......... Epo[ 0][  52/ 771][t69.74] d_l:0.0000 fstol:nan cg32:0 cg64:0 cg128:0 el:nan er:nan no:nan mo:nan g_l:nan gL1:nan(sym:nan) gL2:nan(sym:nan) gL3:nan(sym:nan) gadv:0.0000 dv_l:0.00 iden:nan, tv:nan 
````

**Comment by HRLTY** [comment]
Hi. Maybe check your cross-entropy loss, that might cause NAN if you pass some label incorrectly.
Rui Huang

**Comment by m7schoen** [comment]
Hi, you were right, it was the cross-entropy loss of the identity prediction. The output size (logits) were smaller than the highest id in the dataset. Now everything works as expected. Thanks

**Comment by gavinsz** [comment]
@m7schoen   Do you get deepface model file?  Can you sent it to me?

**Comment by m7schoen** [comment]
@gavinsz I used a LCNN model from a different repository where they shared the weights. https://github.com/yxu0611/Tensorflow-implementation-of-LCNN
I had to replace the model definition to make it work.

**Comment by gavinsz** [comment]
@m7schoen thanks!
Can you share you TP-GAN forked branch?

**Comment by erikgajdos** [comment]
@m7schoen Hello, thanks for your lead, how to make it works without pre-trained weights.
I checked your suggestion to use LCNN from different repository  with model redefinition. Unfortunately I am not able to make it work by my own. I would like to ask you, is it possible for you to commit your code or share it with me, so I will be able to compare original source code and yours updated, so I will understand what model redefinition means?

**Comment by gavinsz** [comment]
@m7schoen  How to get eye, mouth features from image?

**Comment by FUHUACANGYUE** [comment]
@m7schoen 
Can you share your modified TP-GAN code?
